irr/src/CImageWriterJPG.cpp

```diff
@@ -130,32 +131,28 @@ static bool writeJPEGFile(io::IWriteFile *file, IImage *image, u32 quality)
 	jpeg_set_quality(&cinfo, quality, TRUE);
 	jpeg_start_compress(&cinfo, TRUE);
 
-	u8 *dest = new u8[dim.Width * 3];
+	std::vector<u8> dest(dim.Width * 3);
```

Comment:
this is a case for `std::make_unique<u8[]>(dim.Width * 3)` in my opinion; a vector is overkill, we don't need the "dynamically growable" part

Reply:
if should be a std::array and not a std::vector

Reply:
I wasn't sure if a unique pointer or vector fits better; thanks for the feedback.
I have changed it to `std::unique_ptr<u8[]> dest{new u8[dim.Width * 3]};` instead of using `std::make_unique` since the latter redundantly initialises the memory to zero as far as I know.